packages/web3-core/test/unit/web3_subscription.test.ts

```diff
@@ -15,25 +15,32 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { Web3SubscriptionManager } from '../../src';
 import { ExampleSubscription } from './fixtures/example_subscription';
 
+const subscriptions = { example: ExampleSubscription as never };
+
 describe('Web3Subscription', () => {
 	let requestManager: any;
+	let subscriptionManager: Web3SubscriptionManager<any, any>;
 	let sub: ExampleSubscription;
 
 	beforeEach(() => {
-		// const web3 = new Web3('http://localhost:8545');
 		requestManager = {
-			send: jest.fn(),
+			send: jest.fn().mockImplementation(async () => {
+				return 'sub-id';
+			}),
 			on: jest.fn(),
 			provider: { on: jest.fn(), removeListener: jest.fn(), request: jest.fn() },
 		};
-		sub = new ExampleSubscription({ param1: 'value' }, { requestManager });
+		subscriptionManager = new Web3SubscriptionManager(requestManager, subscriptions);
```

Comment:
lgtm, could you also add following integration tests:
- at least two subscriptions, and 
- another test with two subs where one sub is un-subd, and other should be running. 
Thanks